src/daily_output.test.ts

```diff
@@ -0,0 +1,175 @@
+import { expect } from 'chai';
+import { existsSync, rmSync, writeFileSync } from 'fs';
+import Sinon from 'sinon';
+import { OUTPUT_NAME } from './constants';
+import { DailyOutput } from './daily_output';
+import { OutputData } from './inferno_types';
+
+// this JSON is different than the base template
+const mockDailyOutput: OutputData = {
+	lastUpdated: 914200,
+	blockHeight: 914200,
+	timestamp: 1650043449000,
+	PSTHolders: {},
+	wallets: {},
+	ranks: {
+		daily: {
+			hasReachedMinimumGroupEffort: false,
+			groupEffortRewards: [],
+			streakRewards: []
+		},
+		weekly: {
+			hasReachedMinimumGroupEffort: false,
+			groupEffortRewards: [],
+			streakRewards: []
+		},
+		lastWeek: {
+			hasReachedMinimumGroupEffort: false,
+			groupEffortRewards: [],
+			streakRewards: []
+		},
+		total: {
+			groupEffortRewards: [],
+			streakRewards: []
+		}
+	}
+};
+
+// The above JSON stringified with tabs and a trailing new line
+const mockStringifyWithTabsAndTrailingNewLine = `${JSON.stringify(mockDailyOutput, null, '\t')}\n`;
+
+// a valid JSON with missing mandatory fields
+const mockMalformedDailyOutput = {
+	lastUpdated: 914200,
+	blockHeight: 914200,
+	timestamp: 1650043449000,
+	PSTHolders: {},
+	// wallets: {},
+	ranks: {
+		daily: {
+			hasReachedMinimumGroupEffort: false,
+			groupEffortRewards: [],
+			streakRewards: []
+		},
+		weekly: {
+			hasReachedMinimumGroupEffort: false,
+			groupEffortRewards: [],
+			streakRewards: []
+		},
+		lastWeek: {
+			hasReachedMinimumGroupEffort: false,
+			groupEffortRewards: [],
+			streakRewards: []
+		}
+		// total: {
+		// 	groupEffortRewards: [],
+		// 	streakRewards: []
+		// }
+	}
+};
+
+// The above JSON stringified with tabs and a trailing new line
+const mockMalformedStringifyWithTabsAndTrailingNewLine = `${JSON.stringify(mockMalformedDailyOutput, null, '\t')}\n`;
+
+describe('DailyOutput class', () => {
+	const output = new DailyOutput();
+
+	describe('read method', () => {
+		before(() => {
+			if (existsSync(OUTPUT_NAME)) {
+				rmSync(OUTPUT_NAME);
+			}
+		});
+
+		it('return the template if the file is not present', () => {
+			expect(output.readOutputFile).to.throw();
+			expect(output.read()).to.deep.equal({
+				blockHeight: 914117,
+				timestamp: 1650043442070,
```

Comment:
Looks like the template data was just different, this should fix that test:

```suggestion
				timestamp: 1650043442070,
				lastUpdated: 1650043442070,
```

Reply:
Should we add a github action to test/build here on PRs here? I might a branch a PR off of this

Reply:
There a0f7c38d5ce1aff047214750bc21e4c06b49a16f